bin/build.sh

```diff
@@ -3,33 +3,148 @@
+CFLAGS              C compiler flags (defaults to '${PORT_CFLAGSD}')
+CXXFLAGS            C++ compiler flags (defaults to '${PORT_CXXFLAGSD}')
+LDFLAGS             C/C++ linker flags (defaults to '${PORT_LDFLAGSD}')
+PORT_EXTRA_CPPFLAGS C/C++ pre-processor flags to append to CPPFLAGS (defaults to '')
+PORT_EXTRA_CFLAGS   C compiler flags to append to CFLAGS (defaults to '')
+PORT_EXTRA_CXXFLAGS C++ compiler flags to append to CXXFLAGS (defaults to '')
+PORT_EXTRA_LDFLAGS  C/C++ linker flags to append to LDFLAGS (defaults to '')
+PORT_NUM_JOBS       Number of jobs that can be run in parallel (defaults to 1/2 the CPUs on the system)
+PORT_BOOTSTRAP      Bootstrap program to run. If skip is specified, no bootstrap step is performed (defaults to '${PORT_BOOTSTRAPD}')
+PORT_BOOTSTRAP_OPTS Options to pass to bootstrap program (defaults to '${PORT_BOOTSTRAP_OPTSD}')
+PORT_CONFIGURE      Configuration program to run. If skip is specified, no configuration step is performed (defaults to '${PORT_CONFIGURED}')
+PORT_CONFIGURE_OPTS Options to pass to configuration program (defaults to '--prefix=\${PORT_INSTALL_DIR}')
+PORT_INSTALL_DIR    Installation directory to pass to configuration (defaults to '\${HOME}/zot/prod/<pkg>')
+PORT_MAKE           Build program to run. If skip is specified, no build step is performed (defaults to '${PORT_MAKED}')
+PORT_MAKE_OPTS      Options to pass to build program (defaults to '-j\${PORT_NUM_JOBS}')
+PORT_CHECK          Check program to run. If skip is specified, no check step is performed (defaults to '${PORT_CHECKD}') 
+PORT_CHECK_OPTS     Options to pass to check program (defaults to '${PORT_CHECK_OPTSD}')
+PORT_INSTALL        Installation program to run. If skip is specified, no installation step is performed (defaults to '${PORT_INSTALLD}')
+PORT_INSTALL_OPTS   Options to pass to installation program (defaults to '${PORT_INSTALL_OPTSD}')"
+
+}
+
+setDefaults()
+{
+	export PORT_CCD="xlclang"
+	export PORT_CXXD="xlclang++"
+  export PORT_CPPFLAGSD="-DNSIG=9 -D_XOPEN_SOURCE=600 -D_ALL_SOURCE -D_OPEN_SYS_FILE_EXT=1 -D_AE_BIMODAL=1 -D_ENHANCED_ASCII_EXT=0xFFFFFFFF"
+  export PORT_CFLAGSD="-qascii"
+  export PORT_CXXFLAGSD="-+ -qascii"
+	export PORT_BOOSTRAPD="./bootstrap"
+	export PORT_BOOSTRAP_OPTSD=""
+	export PORT_CONFIGURED="./configure"
+	export PORT_MAKED="make"
+	export PORT_CHECKD="make"
+	export PORT_CHECK_OPTSD="check"
+	export PORT_INSTALLD="make"
+	export PORT_INSTALL_OPTSD="install"
+}
+
+printSyntax() 
+{
+  args=$*
+  echo "" >&2
+  echo "build.sh is a general purpose build script to be used with the ZOSOpenTools ports." >&2
+  echo "The specifics of how the tool works can be controlled through environment variables." >&2
+  echo "The only environment variables you _must_ specify are to tell build.sh where the " >&2 
+  echo "  source is, and in what format type the source is stored." >&2
+  echo "By default, the environment variables are defined in a file named setenv in the " >&2 
```

Comment:
```suggestion
  echo "By default, the environment variables are defined in a file named setenv.sh in the " >&2 
```

Reply:
yeah - i'm not quite there yet but yes - intent is to have build.sh source setenv.sh (and step 2 will be to rename setenv.sh to something else - maybe .env although i'm wondering if .bldenv might be better so that people don't get confused?)

Reply:
I do like .buildenv better than setenv.sh or .env.